### ISSUE TITLE
catkin_make fails 

### ISSUE THREAD

**Issue by sh0w** [ravijo/ros_openpose#17]
hi!

we have [openpose-staf](https://github.com/soulslicer/openpose/tree/staf) running on our 
- Jetson Xavier with
- Ubuntu 18.04
- running ROS Melodic 

we are now trying to use your `openpose_ros` to read from a camera stream (image is sent as ROS Message) and send joint info via ros message to another package.

unfortunately 
`catkin_make` fails with this output:
```
rider@rider:~/Desktop/catkin_ws$ catkin_make ros_openpose
Base path: /home/rider/Desktop/catkin_ws
Source space: /home/rider/Desktop/catkin_ws/src
Build space: /home/rider/Desktop/catkin_ws/build
Devel space: /home/rider/Desktop/catkin_ws/devel
Install space: /home/rider/Desktop/catkin_ws/install
####
#### Running command: "cmake /home/rider/Desktop/catkin_ws/src -DCATKIN_DEVEL_PREFIX=/home/rider/Desktop/catkin_ws/devel -DCMAKE_INSTALL_PREFIX=/home/rider/Desktop/catkin_ws/install -G Unix Makefiles" in "/home/rider/Desktop/catkin_ws/build"
####
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Using CATKIN_DEVEL_PREFIX: /home/rider/Desktop/catkin_ws/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/melodic
-- This workspace overlays: /opt/ros/melodic
-- Found PythonInterp: /usr/bin/python2 (found suitable version "2.7.17", minimum required is "2") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python2
-- Using Debian Python package layout
-- Using empy: /usr/bin/empy
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/rider/Desktop/catkin_ws/build/test_results
-- Found gtest sources under '/usr/src/googletest': gtests will be built
-- Found gmock sources under '/usr/src/googletest': gmock will be built
-- Found PythonInterp: /usr/bin/python2 (found version "2.7.17") 
-- Using Python nosetests: /usr/bin/nosetests-2.7
-- catkin 0.7.26
-- BUILD_SHARED_LIBS is on
-- BUILD_SHARED_LIBS is on
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- ~~  traversing 1 packages in topological order:
-- ~~  - ros_openpose
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- +++ processing catkin package: 'ros_openpose'
-- ==> add_subdirectory(ros_openpose)
CMake Error at ros_openpose/CMakeLists.txt:9 (target_link_libraries):
  Cannot specify link libraries for target "ros_openpose_node" which is not
  built by this project.


-- Configuring incomplete, errors occurred!
See also "/home/rider/Desktop/catkin_ws/build/CMakeFiles/CMakeOutput.log".
See also "/home/rider/Desktop/catkin_ws/build/CMakeFiles/CMakeError.log".
Invoking "cmake" failed

```

_edit: this is our CMakeLists in rosopenpose, where we added the path to our openpose and our libopenpose.so_

```


cmake_minimum_required(VERSION 2.8.3)
project(ros_openpose)

## Compile as C++11, supported in ROS Kinetic and newer
add_definitions(-std=c++11)

## Find catkin macros and libraries
find_package(catkin REQUIRED COMPONENTS
  roscpp
  rospy
  cv_bridge
  std_msgs
  sensor_msgs
  image_transport
  message_generation
)

## Make sure 'FindGFlags.cmake' and 'FindGlog.cmake' are visible to cmake
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")



###### begin changes by me in CMakeLists ######
set(OPENPOSE_DIR ~/Desktop/openpose_A_F/openpose-staf)
set(OPENPOSE_INCLUDE_DIRS ${OPENPOSE_DIR}/include)
set(OPENPOSE_LINK_LIBRARY ${OPENPOSE_DIR}/build/src/openpose/libopenpose.so)

target_link_libraries(${PROJECT_NAME}_node
${catkin_LIBRARIES}
${OpenCV_LIBS}
${OPENPOSE_LINK_LIBRARY}
${GFLAGS_LIBRARY}
${GLOG_LIBRARY}
)
##### end changes by me in CMakeLists ######


## Add OpenPose and other dependencies
find_package(GFlags)
find_package(Glog)
find_package(OpenCV REQUIRED)
find_package(OpenPose REQUIRED)
find_package(OpenMP)

if(OPENMP_FOUND)
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
endif()

## Pass in the version of OpenPose
add_definitions( -DOpenPose_VERSION_MAJOR=${OpenPose_VERSION_MAJOR} )
add_definitions( -DOpenPose_VERSION_MINOR=${OpenPose_VERSION_MINOR} )
add_definitions( -DOpenPose_VERSION_PATCH=${OpenPose_VERSION_PATCH} )

## Print OpenPose version in the terminal
MESSAGE(STATUS "OpenPose VERSION: " ${OpenPose_VERSION})

## Declare ROS messages
add_message_files(
  FILES
  Pixel.msg
  BodyPart.msg
  Person.msg
  Frame.msg
)

generate_messages(
  DEPENDENCIES
  std_msgs
  geometry_msgs
)

# catkin specific configuration
catkin_package(
  INCLUDE_DIRS include
  CATKIN_DEPENDS
  std_msgs
  geometry_msgs
  message_runtime
)

## Build
# Specify additional locations of header files
include_directories(
  include
  ${OpenPose_INCLUDE_DIRS}
  ${GFLAGS_INCLUDE_DIR}
  ${GLOG_INCLUDE_DIR}
  ${OpenCV_INCLUDE_DIRS}
  ${catkin_INCLUDE_DIRS}
)

link_directories(
  ${catkin_LIBRARY_DIRS}
)

## Path to Caffe lib i.e., libcaffe.so
## find /home/ravi/tools/openpose -name libcaffe.so
#set(CAFFE_LIB_FOLDER /home/ravi/tools/openpose/build/caffe/lib)
#link_directories(${CAFFE_LIB_FOLDER})

# Declare a C++ executable and
# specify libraries to link a executable target against
add_executable(rosOpenpose
  src/rosOpenpose.cpp
  src/cameraReader.cpp)

# Bug fix
# src: https://github.com/ravijo/ros_openpose/issues/14#issuecomment-631164300
add_dependencies(rosOpenpose ros_openpose_generate_messages_cpp)


target_link_libraries(rosOpenpose
  ${OpenPose_LIBS}
  ${GFLAGS_LIBRARY}
  ${GLOG_LIBRARY}
  ${OpenCV_LIBS}
  ${catkin_LIBRARIES}
)

add_executable(testCameraReader
  src/testCameraReader.cpp
  src/cameraReader.cpp)

target_link_libraries(testCameraReader
  ${OpenPose_LIBS}
  ${GFLAGS_LIBRARY}
  ${GLOG_LIBRARY}
  ${OpenCV_LIBS}
  ${catkin_LIBRARIES}
)
```

any advice on this?

thank you and all the best!

sarah

**Comment by sh0w** [comment]
I fixed this error by adding [findOpenPose.cmake](https://github.com/robotology/human-sensing/blob/master/yarpOpenPose/cmake/Findopenpose.cmake#L15) to `ros_openpose/cmake/Modules`.

at the top of this file i added the line
`set(openpose_ROOT_DIR *path to my openpose*)`

then i changed  `OpenPose_LIBS` to `OpenPose_LIBRARIES` in `target_link_libraries` in `CMakeLists.txt`
so it is now finding my OpenPose.

I also hat to change 
`  ${OpenPose_INCLUDE_DIRS}
`to
`  ${OPENPOSE_INCLUDE_DIRS} 
`

**Comment by ravijo** [comment]
I tested `ros_openpose` with latest OpenPose (master branch) and with OpenPose version 1.6.0. I found that there is no need to set openpose_ROOT_DIR by modifying the CMake file.

Furthermore, a change in the variable `${OpenPose_INCLUDE_DIRS}` is not needed.

I guess you forgot to execute `sudo make install` after compiling OpenPose.

_Please note that I tested these compilations inside fresh Ubuntu 18.04 LTS inside VM._
